public/javascript/posts/doctorLocationFetch.js

```diff
@@ -26,10 +28,11 @@ var app = angular.module('BookAppointmentApp', ['ui.calendar', 'ui.bootstrap', '
           $scope.disableLocation = false;
           $scope.locations = doctorDetails.locations;
           if ($scope.locations.length > 0) {
-            $scope.locations.push({ 'id': -1, 'name': 'All' });
-            $scope.locationId = $scope.locations[0].id;
+            if($scope.locations[$scope.locations.length - 1]['id'] !== -1)
```

Comment:
Using underscore you can check in hash if id is present ,
suppose you want to use underscore method _.pluck then you will use it here like - $scope.pluck